core/shared/src/main/scala-3/zio/ZLayerCompanionVersionSpecific.scala

```diff
@@ -38,4 +40,24 @@ trait ZLayerCompanionVersionSpecific {
    */
   def makeSome[R0, R] =
     new WireSomePartiallyApplied[R0, R]
+
+  /**
+   * Derives a simple layer for a case class given as a type parameter.
+   * {{{
+   * case class Car(engine: Engine, wheels: Wheels)
+   * val derivedLayer: ZLayer[Engine & Wheels, Nothing, Car] = ZLayer.deriveLayer[Car]
+   * // equivalent to:
+   * val manualLayer: ZLayer[Engine with Wheels, Nothing, Car] = ZLayer {
```

Comment:
We should update this example to reflect the idiomatic way of doing this manually.

Reply:
You mean the `ZLayer#fromFunction`, right? I'll do.